### PR TITLE
Smart organ storage fix

### PR DESCRIPTION
**Pull request by norill**
## About The Pull Request
This fixes a few goofs in smart organ storage code, namely:
- return instead of continue in a loop, which broke the organ repairing functionality
- not factored in maxHealth, which resulted in absurdly slow repair rates
- repair rate used STANDARD_ORGAN_HEALING divided by 2, while organ/proc/on_life uses STANDARD_ORGAN_HEALING multiplied by up to 4

The repair rate is still slow as fuck even when upgraded with bluespace matter bin and should probably be buffed, but thats none of my business, im just fixing shit here

## Why It's Good For The Game
Organ repairing feature of Smart organ storage is broken, which makes upgrading it kinda pointless

## Changelog
:cl:
fix: fixed buggy organ fridges repair rate
fix: fixed upgraded organ fridges not repairing organs
/:cl:

### CODE REVIEW

File: code/__DEFINES/DNA.dm [hunk 1]
```diff
@@ -166,7 +166,7 @@
 
 //organ defines
 #define STANDARD_ORGAN_THRESHOLD 100
-#define STANDARD_ORGAN_HEALING 0.0005
+#define STANDARD_ORGAN_HEALING 0.00050
```
Comment (ATH1909) [inline code/__DEFINES/DNA.dm:169]: ?

**Comment by ATH1909** [comment]
> while organ/proc/on_life uses STANDARD_ORGAN_HEALING multiplied by up to 4

can't the healing from healing_factor technically be multiplied by up to 5 due to high satiety? also, remember that having a nonzero satiety is unusual, not the norm (IIRC).

```
/obj/item/organ/proc/on_life(delta_time, times_fired) //repair organ damage if the organ is not failing
	if(organ_flags & ORGAN_FAILING)
		handle_failing_organs(delta_time)
		return

	if(failure_time > 0)
		failure_time--

	if(organ_flags & ORGAN_SYNTHETIC_EMP) //Synthetic organ has been emped, is now failing.
		applyOrganDamage(decay_factor * maxHealth * delta_time)
		return
	///Damage decrements by a percent of its maxhealth
	var/healing_amount = healing_factor
	///Damage decrements again by a percent of its maxhealth, up to a total of 4 extra times depending on the owner's health
	healing_amount += (owner.satiety > 0) ? (4 * healing_factor * owner.satiety / MAX_SATIETY) : 0
	applyOrganDamage(-healing_amount * maxHealth * delta_time, damage) // pass curent damage incase we are over cap
```

File: code/__DEFINES/DNA.dm (hunk 1 above)
Comment (norill) [reply code/__DEFINES/DNA.dm:169]: this way it's clearer to see that it's 50/100 000, not 5/100 000. i was comparing that value with STANDARD_ORGAN_DECAY 0.00111 and at first glance i thought the ratio of decay to healing was 111/5, but in reality it's 111/50

**Comment by norill** [comment]
> 
> 
> > while organ/proc/on_life uses STANDARD_ORGAN_HEALING multiplied by up to 4
> 
> can't the healing from healing_factor technically be multiplied by up to 5 due to high satiety? also, remember that having a satiety >0 is unusual, not the norm (IIRC).
> 

shit, ur right. i didnt notice the + in ```healing_amount +=```. i guess imma revert that line, it might not have been an oversight after all, could be intended that way

File: code/__DEFINES/DNA.dm (hunk 1 above)
Comment (Mothblocks) [reply code/__DEFINES/DNA.dm:169]: Can you just put in `(50 / 100000)` then? This isn't super readable.

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 2]
```diff
@@ -452,8 +452,8 @@
 	for(var/organ in contents)
 		var/obj/item/organ/O = organ
 		if(!istype(O))
-			return
-		O.applyOrganDamage(-repair_rate * delta_time)
+			continue
+		O.applyOrganDamage(-repair_rate * O.maxHealth * delta_time)
```
Comment (Mothblocks) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm:456]: Turn this into `for (var/obj/item/organ/organ in contents)` and remove the istype.

**Review by Mothblocks — CHANGES_REQUESTED** [review]

**Pull request merged**